xbmc/threads/Thread.cpp

```diff
@@ -243,3 +255,69 @@ void CThread::Action()
   }
 }
 
+void CThread::SetCrashProtection(IThreadCrashProtection* crashProtection)
+{
+  if (g_application.IsCurrentThread())
+    g_application.SetCrashProtection(crashProtection);
```

Comment:
Maybe I can add it to `IThreadCrashProtection` itself and the call it from there to `CThread` and `CApplication`.